openmls/src/treesync/mod.rs

```diff
@@ -234,12 +249,9 @@ impl TreeSync {
     pub(crate) fn from_nodes_without_leaf(
         backend: &impl OpenMlsCryptoProvider,
         ciphersuite: &Ciphersuite,
-        node_options: &[Option<Node>],
+        mut nodes: Vec<Option<Node>>,
```

Comment:
This PR might fix [#652](https://github.com/openmls/openmls/issues/652) as well.

Reply:
It's a step in towards fixing it. But `from_nodes` above is still using a slice.